src/NUnitFramework/framework/Internal/Execution/WorkItem.cs

```diff
@@ -248,6 +248,16 @@ public virtual void Execute()
                     Test.RequiresThread ? "RequiresThread" : "different Apartment");
 
 #if APARTMENT_STATE
+#if NETSTANDARD2_0
+                if (!OSPlatform.CurrentPlatform.IsWindows)
+                {
+                    string msg = "Apartment state cannot be set on non-windows platforms.";
```

Comment:
Since we aren't checking `WorkQueue.TargetApartment != ApartmentState.MTA` before checking if we're on Windows, this means that no workers are started.

https://github.com/nunit/nunit/blob/cbeffad907943d4219a2f0cdcea55830d267225e/src/NUnitFramework/framework/Internal/Execution/ParallelWorkItemDispatcher.cs#L169-L172

I think this is the reason that the CI builds are hanging when not on Windows.

A good question is whether we should be setting the apartment state to MTA at all or leaving it Unknown and skipping the call to SetApartmentState.